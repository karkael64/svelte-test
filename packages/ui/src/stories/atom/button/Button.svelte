<script lang="ts">
  import { createEventDispatcher } from 'svelte';

  export let primary = false;
  export let size = 'medium';
  export let label: string;

  let mode = primary ? 'storybook-button--primary' : 'storybook-button--secondary';

  const dispatch = createEventDispatcher();

  function onClick(event: Event) {
    dispatch('click', event);
  }
</script>

<button
  type="button"
  class={`storybook-button storybook-button--${size} ${mode}`}
  on:click={onClick}>
  {label}
</button>

<style>
  .storybook-button {
    font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
    font-weight: 700;
    border: 0;
    border-radius: 3em;
    cursor: pointer;
    display: inline-block;
    line-height: 1;
  }
  .storybook-button--primary {
    color: white;
    background-color: #1ea7fd;
  }
  .storybook-button--secondary {
    color: #333;
    background-color: transparent;
    box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
  }
  .storybook-button--small {
    font-size: 12px;
    padding: 10px 16px;
  }
  .storybook-button--medium {
    font-size: 14px;
    padding: 11px 20px;
  }
  .storybook-button--large {
    font-size: 16px;
    padding: 12px 24px;
  }
</style>