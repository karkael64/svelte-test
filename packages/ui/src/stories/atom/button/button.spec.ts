import { fireEvent } from "@testing-library/svelte";
import Button from "./Button.svelte";
import { renderWithEvents } from "../../../../test/renderWithEvents";

describe("Button", () => {
  it("Should instanciate with default values", async () => {
    const label = "inner text";
    const { getByText, container } = renderWithEvents(Button, {
      label,
    });

    const button = getByText(label);
    expect(button).toBeInTheDocument();

    expect(button).toHaveClass("storybook-button");
    expect(button).toHaveClass("storybook-button--medium");
    expect(button).toHaveClass("storybook-button--secondary");

    expect(container).toMatchSnapshot();
  });

  it("Should change size to small", async () => {
    const label = "inner text";
    const { getByText } = renderWithEvents(Button, {
      label,
      size: "small",
    });

    const button = getByText(label);
    expect(button).toHaveClass("storybook-button--small");
  });

  it("Should change size to large", async () => {
    const label = "inner text";
    const { getByText } = renderWithEvents(Button, {
      label,
      size: "large",
    });

    const button = getByText(label);
    expect(button).toHaveClass("storybook-button--large");
  });

  it("Should change size to primary", async () => {
    const label = "inner text";
    const { getByText } = renderWithEvents(Button, {
      label,
      primary: true,
    });

    const button = getByText(label);
    expect(button).toHaveClass("storybook-button--primary");
  });

  it("Should trigger click", async () => {
    const label = "inner text";
    const handleClick = jest.fn();
    const { getByText } = renderWithEvents(
      Button,
      { label },
      { click: handleClick }
    );

    const button = getByText(label);
    await fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
