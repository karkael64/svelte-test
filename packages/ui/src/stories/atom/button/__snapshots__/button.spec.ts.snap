// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should instanciate with default values 1`] = `
<body>
  <div>
    <button
      class="storybook-button storybook-button--medium storybook-button--secondary svelte-fztlcy"
      type="button"
    >
      inner text
    </button>
  </div>
</body>
`;
